BTCPayServer/Controllers/UIInvoiceController.UI.cs

```diff
@@ -198,6 +198,9 @@ public async Task<IActionResult> InvoiceReceipt(string invoiceId, [FromQuery] bo
             var store = await _StoreRepository.GetStoreByInvoiceId(i.Id);
             if (store is null)
                 return NotFound();
+            
+            if (i.Archived && User.Identity is { IsAuthenticated: false })
+                return StatusCode(403);
```

Comment:
I think it should rather be something like this, so that we check the actual permission and return a 404 (like we do elsewhere in such cases):

```suggestion
            if (i.Archived && !(await _authorizationService.AuthorizeAsync(User, i.StoreId, Policies.CanViewInvoices)).Succeeded)
                return NotFound();
```

Reply:
Thanks for the suggestion! I've applied the changes as proposed. Let me know if there's anything else to adjust.